base/expr.jl

```diff
@@ -464,7 +464,10 @@ macro generated(f)
                          Expr(:block,
                               lno,
                               Expr(:if, Expr(:generated),
-                                   body,
+                                   # https://github.com/JuliaLang/julia/issues/25678
+                                   Expr(:block,
+                                        :(local tmp = $body),
+                                        :(if tmp isa Core.CodeInfo; return tmp; else tmp; end)),
```

Comment:
@simeonschaub I guess generally it's a bad idea to name a module `Core` but this caused a bug in Turing: https://github.com/TuringLang/Turing.jl/issues/1756#issuecomment-1012622944 and discussion in the issue. I think this could be fixed by interpolating `Core.CodeInfo`. I could prepare a PR (my first to base actually :open_mouth:) if you consider this to be a bug that should be fixed in base (I'll submit a PR to Turing in any case that renames the offending module).

Reply:
That's funny, I was just staring at code generated by this and was wondering whether having a call to `getproperty` in here was the right thing! It's generally more idiomatic to handle this as `$(GlobalRef(Core, :CodeInfo))` instead of just interpolating the object though since that's what lowering would generate.

Reply:
Ref: https://github.com/JuliaLang/julia/pull/43823